src/runtime/d3d12compute.cpp

```diff
@@ -3023,8 +3012,9 @@ WEAK int halide_d3d12compute_run(void *user_context,
     StartCapturingGPUActivity();
 #endif
 
-    halide_assert(user_context, state_ptr);
-    module_state *state = (module_state *)state_ptr;
+    d3d12_library *library{};
```

Comment:
nit: I know this zero-inits, but IMHO using `= nullptr` is more idiomatic for a pointer type.

Reply:
"{}" says to the compiler "default init this thing, no matter what it is." Given there's push back to do this consistently in lots of places, I'd say both the concision and that it works for all types, even in generic code, is a strong reason to prefer this construct. The only reason initialization is needed in this code is if asserts do not end the computation. Moreover, a lot of this would go away with an Optional type.